Plugins/Creature/Creature.cpp

```diff
@@ -1876,4 +1882,46 @@ ArgumentStack Creature::SetDisarmable(ArgumentStack&& args)
     }
     return Services::Events::Arguments();
 }
+
+ArgumentStack Creature::SetFaction(ArgumentStack&& args)
+{
+    int32_t retVal = -1;
+    if (auto *pCreature = creature(args))
+    {
+        const auto factionid = Services::Events::ExtractArgument<int32_t>(args);
+        auto* pFaction= Globals::AppManager()->m_pServerExoApp->m_pcExoAppInternal->m_pFactionManager->GetFaction(factionid);
+        if (pFaction)
+        {
+            pFaction->AddMember(pCreature->m_idSelf);
+            retVal = 1;
+        }
+        else
+        {
+            LOG_NOTICE("NWNX_Creature function called with invalid faction id");
+        }
+    }
+    else
+    {
+        LOG_NOTICE("NWNX_Creature function called on invalid creature");
+    }
+    return Services::Events::Arguments(retVal);
+}
+
+
+ArgumentStack Creature::GetFaction(ArgumentStack&& args)
+{
+    int32_t retVal = -1;
+    if(auto *pCreature = creature(args))
+    {
+        auto *faction = pCreature->GetFaction();
```

Comment:
I'd change this to `if (auto *pFaction = pCreature->GetFaction())` on the off chance GetFaction() returns a nullptr.